onnxscript/rewriter/_ir_utils.py

```diff
@@ -124,3 +124,17 @@ def has_rank(value: ir.Value | None, rank: int) -> bool:
         return False
     shape = value.shape
     return (shape is not None) and (shape.rank() == rank)
+
+
+def get_dim(value: ir.Value | None, dim: int) -> ir.SymbolicDim | int | None:
+    """Returns the value of the given dimension, or None if it is not statically known."""
+    if value is None:
+        return None
+    shape = value.shape
+    if shape is None:
+        return None
+    if dim < 0:
+        dim += shape.rank()
```

Comment:
Can this be absorbed into `return shape[dim]`?

Reply:
You mean that this is already handled by `shape[dim]` ? I suppose so, but we would need to adjust the condition in the next line to `dim < -shape.rank()` (to avoid an exception in the indexing). I find this as an easy way to remember the condition to check (whether it is < or <= etc.)

Reply:
I see!